### PR TITLE
 Fixing Failing Test Suites for Patient and Order Entities

### PR DESCRIPTION
**Pull request by manishjha-04**
>Fixing Issue #787 

**Previously:**
Screenshot 2024-03-10 235508

**Currently:**
Screenshot 2024-03-10 234326

**>Recording Before:**

https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/128405481/28f916db-c6f8-4618-85b0-ab62e48f236a

https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/128405481/28d601f6-16fd-4ace-b14d-b46ad6b0845b

.

**>Recording After:**

https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/128405481/1c916854-2912-4204-9c57-d57ffb098614

https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/128405481/64b002d2-ad59-4a85-9354-b38edf0b0c1a

### CODE REVIEW

**Comment by manishjha-04** [comment]
Hey @mozzy11    please review  and let me know if any changes needed!

**Comment by manzil-infinity180** [comment]
> Thats great @manishjha-04
> 
> if possible can you demonstrate the same considering your local machine
> 
> i.e instead of using https://oe3.openelis-global.org/ can you perform the same on `localhost`
> 
> and capture the video IG then it will not work
> 
> just a suggestion, its upto you

all the test that was present yet is using that https://oe3.openelis-global.org/  so i think no need to check with the localhost

**Comment by manzil-infinity180** [comment]
> Hey @codewithunknown i do agree with @manzil-infinity180 since the predefined test url was for OpenELIS website(and not the dev one), so i dont think there is a need but if you want it i can give it a try and share the video!

@manishjha-04 just going through cypress , what @codewithunknown said is correct think if the code is not merged yet we have to look for that into the localhost to check, the point is at the last we have to check from localhost

**Comment by manishjha-04** [comment]
Hey @manzil-infinity180, I understand your perspective, and I agree with you. What I meant was that I was simply refining existing entities. However, if there's a requirement for implementation on localhost, I'll certainly address it and adjust the test suites accordingly. In this PR i have just improved which was not working earlier!

File: frontend/package.json [hunk 1]
```diff
@@ -74,6 +74,8 @@
     "**/@types/react": "16.14.5"
   },
   "devDependencies": {
+    "cypress": "^12.17.3",
+    "cypress-xpath": "^2.0.1",
```
Comment (sherrif10) [inline frontend/package.json:78]: Is this the latest version of cypress ?

File: frontend/cypress/e2e/patientEntry.cy.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ describe('Patient Search', function () {
         cy.wait(200)
         patientPage.clickSearchPatientButton();
         cy.fixture('Patient').then((patient) => {
-            patientPage.validatePatientSearchTable(patient.firstName, patient.inValidName);
+            patientPage.validatePatientByGender("M");
         });
         cy.wait(200).reload();
     });
```
Comment (sherrif10) [inline frontend/cypress/e2e/patientEntry.cy.js:49]: For better testing approach, we need to add a configuration that will have to trigger these tests automatically once the PR is pushed.  That might quicken both e2e tests and other work. cc @mozzy11

Comment (tendomart) [reply frontend/cypress/e2e/patientEntry.cy.js:49]: I agree , have we used this [github action](https://github.com/cypress-io/github-action)

@mozzy11

**Comment by sherrif10** [comment]
No we havent i think , Do you mind investigating into it. @tendomart

**Comment by tendomart** [comment]
I will find sometime later and look into it .

**Comment by mozzy11** [comment]
Thanks @manishjha-04 
i added those commits to retrigure all the action builds

**Pull request merged**